src/python/gudhi/weighted_rips_complex.py

```diff
@@ -0,0 +1,59 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Raphaël Tinarrage, Yuichi Ike, Masatoshi Takenouchi
+#
+# Copyright (C) 2020 Inria, Copyright (C) 2020 FUjitsu Laboratories Ltd.
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+from gudhi import SimplexTree
+
+class WeightedRipsComplex:
+    """
+    Class to generate a weighted Rips complex from a distance matrix and weights on vertices, 
+    in the way described in the paper 'DTM-based filtrations' https://arxiv.org/abs/1811.04757.
```

Comment:
Note that we have a file `biblio/bibliography.bib` at the root of gudhi, and you can use it with ``:cite:`the_label` `` instead of giving the URL.